x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -478,114 +473,32 @@ export class ObservabilityAIAssistantClient {
       simulateFunctionCalling?: boolean;
       tracer: LangTracer;
     }
```

Comment:
Maybe a premature optimization but perhaps we should extract the `complete` function. The goal should be improving readability of the `ObservabilityAIAssistantClient` (the complete function is pretty huge) and also making it easier to remove down the line.